godtools/Managers/TranslationZipImporter.swift

```diff
@@ -152,8 +149,52 @@ class TranslationZipImporter {
         try data.write(to: URL(fileURLWithPath: "\(documentsPath)/\(filename)"))
     }
     
-    private func extractZipFile(_ filename: String) {
-        SSZipArchive.unzipFile(atPath: "\(documentsPath)/\(filename)", toDestination: resourcesPath)
+    private func extractZipFile(_ filename: String, translationId: String) {
+        let tempDirectoryPath = "\(documentsPath)/\(filename.replacingOccurrences(of: ".zip", with: ""))"
+        do {
+            try FileManager.default.createDirectory(atPath: tempDirectoryPath,
+                                                    withIntermediateDirectories: true,
+                                                    attributes: nil)
+            
+            SSZipArchive.unzipFile(atPath: "\(documentsPath)/\(filename)", toDestination: tempDirectoryPath)
+            
+            try recordReferencedFiles(directoryPath: tempDirectoryPath, translationId: translationId)
+            
+            try moveFilesFrom(directoryPath: tempDirectoryPath)
+            saveToDisk()
+        } catch {
+            Crashlytics().recordError(error, withAdditionalUserInfo: ["customMessage": "Error extracting zip file \(filename)"])
+            rollbackContext()
+        }
+    }
+    
+    private func recordReferencedFiles(directoryPath: String, translationId: String) throws {
+        guard let translation = findEntity(Translation.self, byAttribute: "remoteId", withValue: translationId) else {
+            return
+        }
+        
+        let files = try FileManager.default.contentsOfDirectory(atPath: directoryPath)
+        
+        for filename in files {
+            let referencedFile = findEntity(ReferencedFile.self, byAttribute: "filename", withValue: filename) ?? createEntity(ReferencedFile.self)
+            referencedFile?.filename = filename
+            referencedFile?.addToTranslations(translation)
+        }
+    }
+    
+    private func moveFilesFrom(directoryPath: String) throws {
```

Comment:
moving files from a temp directory to a permanent one is covered in http://nshipster.com/nstemporarydirectory/ as well